File: Assets/MakeHuman/Icons/KsSoft/Editor/Multilingual/tools/protocol/t_SoundParamHeader.py
from CVariable import CReadVariable,CWriteVariable
from Vector import *
from t_SoundParam import	t_SoundParam
#/==========================================================================
#*!
#	@brief	t_SoundParamHeader
#/
class t_SoundParamHeader:
	#/==========================================================================
	#*!
	#	@brief	Member
	#/
	MAGIC_NO = 4276948992
	m_magicNo = 4276948992	#U32
	m_aSoundParam = []	#t_SoundParam
	#/==========================================================================
	#*!
	#	@brief	Constructor
	#/
	def __init__(self):
		self.clear()

	#/==========================================================================
	#*!
	#	@brief	Accessor
	#/
	def clear(self):
		self.m_magicNo = 4276948992
		self.m_aSoundParam = []

	def read(self,cVariable):
		self.m_magicNo = cVariable.getU32()
		n = cVariable.getU16()
		if (n > 65535) :
			cVariable.error('array size error in m_aSoundParam')
			return False
		self.m_aSoundParam = [None]*n
		for i in range(n):
			self.m_aSoundParam[i] = t_SoundParam()
			if (not self.m_aSoundParam[i].read(cVariable)):
				return False
		return True

	def write(self,cVariable):
		cVariable.putU32(self.m_magicNo)
		if (len(self.m_aSoundParam) > 65535) :
			cVariable.error('array size error in m_aSoundParam')
			return False
		n = len(self.m_aSoundParam)
		cVariable.putU16(n)
		for i in range(n):
			if (not self.m_aSoundParam[i].write(cVariable)):
				return False
		return True

File: Assets/MakeHuman/Icons/KsSoft/Plugins/Protocol/t_SoundParamHeader.cs
//==============================================================================================
/*!
	@file  KsSoft Protocol
	
	(counter SJIS string 京)
*/
//==============================================================================================
using UnityEngine;
using System.Collections;
using System.Collections.Generic;

using KS;
//==========================================================================
/*!
	@brief	t_SoundParamHeader
*/
public class t_SoundParamHeader : ISerializable {
	//==========================================================================
	/*!
		@brief	Member
	*/
	public  const uint	MAGIC_NO = 4276948992;	// 4276948992
	public uint	m_magicNo;
	public t_SoundParam[]	m_aSoundParam;
	//==========================================================================
	/*!
		@brief	Constructor
	*/
	public t_SoundParamHeader() {
		clear();
	}
	//==========================================================================
	/*!
		@brief	Accessor
	*/
	public void clear() {
		m_magicNo = 4276948992;	// MAGIC_NO;
		m_aSoundParam = new t_SoundParam[0];
	}
	public bool read(CReadVariable cVariable) {
		try {
			m_magicNo = cVariable.getU32();
			{
				int n = (int) cVariable.getU16();
				if (n > 65535) {
					cVariable.error("array size error in m_aSoundParam");
					return false;
				}
				m_aSoundParam = new t_SoundParam [n];
				for(int i = 0;i < n;++i) {
					m_aSoundParam[i] = new t_SoundParam();
					if (!m_aSoundParam[i].read(cVariable)) return false;
				}
			}
		} catch (System.Exception e) {
			Debug.LogError(e);return false;
		}
		return true;
	}
	public bool write(CWriteVariable cVariable) {
		cVariable.put(m_magicNo);
		{	
				if (m_aSoundParam.Length > 65535) {
					cVariable.error("array size error in m_aSoundParam");
					return false;
				}
			int n = m_aSoundParam.Length;
			cVariable.put((ushort) n);
			for(int i = 0;i < n;++i) {
				if (!m_aSoundParam[i].write(cVariable)) return false;

			}
		}
		return true;
	}
};

